embulk-core/src/main/java/org/embulk/spi/util/Reporters.java

```diff
@@ -0,0 +1,78 @@
+package org.embulk.spi.util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.embulk.spi.Reporter;
+import org.embulk.spi.ReporterCloseable;
+
+import java.util.Map;
+
+public final class Reporters
+        implements AutoCloseable
+{
+    public enum Type
```

Comment:
I was afraid that the name `Type` could be confusing because all "plugins" have `"type"` by default (e.g. `type: null`, `type: stdout`, `type: file`, ...). What do you think: `Channel`, `Target`, `Purpose` or else instead of `Type`?

In addition, should this `enum` is to be in this `Reporters` class?